chrome/content/zotero/elements/attachmentsBox.js

```diff
@@ -113,31 +113,13 @@
 				if (attachments.length === 0 && action !== "delete") {
 					return;
 				}
-				if (action == 'add') {
-					for (let attachment of attachments) {
+				for (let attachment of attachments) {
+					this.querySelector(`attachment-row[attachment-id="${attachment.id}"]`)
+						?.remove();
+					if (action !== 'delete') {
```

Comment:
We should list `modify` and `refresh` explicitly, since others could conceivably be added in the future.